async/src/main/java/org/odk/collect/async/Scheduler.kt

```diff
@@ -77,6 +77,11 @@ interface Scheduler {
     fun cancelAllDeferred()
 
     fun <T> flowOnBackground(flow: Flow<T>): Flow<T>
+
+    enum class NetworkType {
```

Comment:
I was thinking that it's really a constraint so it can be on one of the two network types or `null`. Would you prefer it if the param on `networkDeferred` was `networkConstraint` (and it takes an optional `NetworkType` still) instead of `networkType`?

Reply:
yeah maybe it would make more sense.